force-app/main/default/classes/CompareByStudentName.cls

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+* Avenoir Technologies Pvt. Ltd. All rights reserved.
+* Ticket Number: AVEBLOG79
+--------------------------------------------------------------------------------
+* Purpose/Methods: compare the StudentRecord by Name.
+* Utility Test Class: CompareByStudentNameTest.cls
+--------------------------------------------------------------------------------
+* Version History:
+* VERSION     DEVELOPER NAME      DATE       DETAIL FEATURES
+* 1.0         Satish Yadav      05/02/2024   Initial Development
+* ****************************************************************************/
+public with sharing class CompareByStudentName implements Comparator<StudentRecord> {
+
+    /**
+    * Constructor Purpose: Compares two StudentRecord instances based on the student's 
+      name to determine their order. 
+    * @param StudentRecord : name1
+    * @param StudentRecord : name2
+    */
+
+    public Integer compare(StudentRecord name1, StudentRecord name2) {
```

Comment:
update name1 and name2 to studentOne and StudentTwo

Reply:
Integer returnValue = -1;
        if(String.isNotBlank(name1.studentName)) {
            returnValue = String.isBlank(name2.studentName) ? 1 : name1.studentName.compareTo(name2.studentName);
        } 
        return returnValue;

Reply:
Refactor method like above